### DISCUSSION TITLE
Bill of materials for Bucket 4J

### DISCUSSION THREAD

**Discussion by ghoshrahul** [vladimir-bukhtoyarov/bucket4j#526] [Q&A]
I wanted to understand whether there is a BOM for Bucket 4J, or do we have to manually import each and every dependency within the pom file.

If there is a BOM, please point the relevant document.

Let me know if there are any clarifications

**Comment by vladimir-bukhtoyarov** [comment]
Hello @ghoshrahul 

0. There are no BOM, because it is not necessary.
1. bucket4j-core does not depend from anything. 
2. According to integration with third-party technologies - each particular integration like `bucket4j-xxx` declares all dependencies with `runtime` scope , usually you should not add something to your project other than bucket4j's related jars, because it is obviously that jars related to `xxx` technology is already configured for your project.
